docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+
+The `ListCommand` class performs the following key operations:
+
+* ListCommand#execute() — Generates a formatted message displaying all events in the list.
+
+These operations are accessible through the `Command` and can be invoked when the list command is entered by the user.
+
+#### Feature Implementation
+
+Given below is an example usage scenario and the behavior of the list feature at each step:
+
+1. User Command Input:
+The user enters the command `list` to view all scheduled events.
+
+2. Command Recognition:
+The `COMMAND_WORD` is set to "list", enabling the system to recognize the command input and invoke `ListCommand`.
+
+3. Execution of ListCommand#execute():
+The `execute()` method retrieves each event from `eventList` and appends it to a formatted output message.
+It uses the `String.format` method with `LIST_MESSAGE` to include the total number of events in the message header.
+Events are appended to `outputMessage` with numbered formatting for readability.
+
+4. Output Generation:
+The method stores the generated `outputMessage` in `this.message`, ready for display.
+
+### Add feature
+
+The `add` feature allow users to add events/participants based on relevant flags.
+It is implemented in the `AddCommand` class which extends the base `Command` class and parse through the command to retrieve information based off flags.
+
+The above operation is implemented as `AddCommand#execute()`. This overrides the `Command#execute()` operation in `Command`,
+and is invoked when the latter operation is called.
+
+The `AddCommand` handles two primary functions:
+
+1. **Add an Event:** When provided with event details, including name, time, venue and priority, it creates and stores a new event in the event list.
+2. **Add a Participant:** When provided with participant information, including name, contact number, and email, it attempts to add the participant to an existing event.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `add` mechanism, and how it behaves at each step.
+
+1. The user enters the command `add` followed by `-e` or `-p` to indicate adding an event or participant.
+2. This step is determined by our `Parser` which parses through the user input to determine if it is adding a participant or event
+3. Based on the parsed input, `AddCommand` executes one of the following actions:
+   + **Add Event:** Creates a new event in `EventList` with the provided event details
+   + **Add Participant:** Locates the event in `EventList` and adds the participant to it
+4. If a duplicate event is found, `DuplicateDataException` is thrown.
+
+The interactions between components during the execution of the `add` command are show in the **Sequence Diagram** below:
+
+**Add Event**
+
+<img src = "images/AddEventSequenceDiagram.png">
+
+**Add Participant**
+
+<img src = "images/AddParticipantSequenceDiagram.png">
```

Comment:
The sequence diagram font size is too small relative to the font size of the main text. Perhaps you can consider making the two font sizes match?
![image](https://github.com/user-attachments/assets/5ab73998-a942-4b75-9636-6f568f01f505)

Reply:
We believe it's more of an issue of the sequence diagram being too large and complicated. We're planning to redo the diagram (other comments pointed out more pressing issues) so we'll take your point into account.